partiql-types/src/main/kotlin/org/partiql/types/StaticType.kt

```diff
@@ -509,6 +516,19 @@ sealed class NumberConstraint {
     }
 }
 
+/**
+ * Represents Tuple constraints this is still; experimental
+ * and subject to change upon finalization of the following:
+ * - https://github.com/partiql/partiql-spec/issues/49
+ * - https://github.com/partiql/partiql-docs/issues/37
+ */
+sealed class TupleConstraint {
```

Comment:
I think the side note is probably the most important aspect of this. If constraints do not get propagated to Views, there wouldn't be a need to introduce the Table/Column constraints directly to the definition of StructType -- as its definition wouldn't be relevant.

While I'm not sure if the following should be how we approach this problem, [Postgres seemingly drops the constraints](https://www.db-fiddle.com/f/75paVmvQp8mPwoz7ryJigV/0) in the definition of the view. If we don't propagate the constraints, then the definition of Table Metadata (or Table Schema) with its constraints and Column Metadata (or Column Schema) with its constraints might lie outside of the definition of StaticType.

Reply:
Just a side note: it looks like before you've completed your review, I've already removed the `PrimaryKey` and `PartitionKey`.

Regarding clashing constraints, I _think_ "weird behavior" that you've pointed out holds some specification assumptions. On the other hand, we still don't have a specification for our type constraints as part of our `Type` and `Schema` pending spec. work. As part the undone work we still require to define constraints in more details and be able to answer the questions like: if constraints can get repeated like [ISL](https://amazon-ion.github.io/ion-schema/docs/isl-2-0/spec#constraints) and if so what is the expected behavior in case of conflicts. The fact is in this PR we're having just enough definition at the moment and intentionally excluding other implementation details to see how this model plays out.

Reply:
> I think the side note is probably the most important aspect of this. If constraints do not get propagated to Views, there wouldn't be a need to introduce the Table/Column constraints directly to the definition of StructType -- as its definition wouldn't be relevant.

I think we still need the constraints on each `StaticType`; the view is just one use-case, some other use-cases are 1) data filtering/cleansing in a sort of DML-like operations and 2) defining the behavior of some of PartiQL core functions like `TUPLEUNION` in the presence of constraints (E.g. `OrderedAttrs`).